metrics/metrics.go

```diff
@@ -0,0 +1,156 @@
+package metrics
+
+import (
+	"fmt"
+	"sync"
+	"time"
+
+	"github.com/DataDog/datadog-go/v5/statsd"
+	"github.com/coopnorge/go-datadog-lib/v2/errors"
+	"github.com/coopnorge/go-datadog-lib/v2/internal"
+)
+
+var (
+	setupOnce    sync.Once
+	setupErr     error
+	statsdClient statsd.ClientInterface
+	errorHandler errors.ErrorHandler
+	cfg          *config
+)
+
+// GlobalSetup configures the Dogstatsd Client. GlobalSetup is intended to be
+// called from coopdatadog.Start(), but can be called directly.
+func GlobalSetup(options ...Option) error {
+	setupOnce.Do(func() {
+		if internal.IsDatadogDisabled() {
+			statsdClient = &noopClient{}
+			return
+		}
+
+		cfg, setupErr = resolveConfig(options)
+		if setupErr != nil {
+			return
+		}
+
+		if !cfg.enableMetrics {
+			statsdClient = &noopClient{}
+			return
+		}
+
+		statsdClient, setupErr = statsd.New(cfg.dsdEndpoint, statsd.WithTags(cfg.tags))
+		if setupErr != nil {
+			return
+		}
+	})
+	return setupErr
+}
+
+// Flush forces a flush of all the queued dogstatsd payloads.
+func Flush() error {
+	err := statsdClient.Flush()
+	if err != nil {
+		return fmt.Errorf("failed to flush: %w", err)
+	}
+	return nil
+}
+
+// Gauge measures the value of a metric at a particular time.
+func Gauge(name string, value float64, tags ...string) {
+	err := statsdClient.Gauge(name, value, tags, cfg.metricSampleRate)
+	if err != nil {
+		errorHandler(fmt.Errorf("failed to send Gauge: %w", err))
+	}
+}
+
+// GaugeWithTimestamp measures the value of a metric at a given time.
+// BETA - Please contact our support team for more information to use this feature: https://www.datadoghq.com/support/
+// The value will bypass any aggregation on the client side and agent side, this is
+// useful when sending points in the past.
+//
+// Minimum Datadog Agent version: 7.40.0
+func GaugeWithTimestamp(name string, value float64, timestamp time.Time, tags ...string) {
+	err := statsdClient.GaugeWithTimestamp(name, value, tags, cfg.metricSampleRate, timestamp)
+	if err != nil {
+		errorHandler(fmt.Errorf("failed to send GaugeWithTimestamp: %w", err))
+	}
+}
```

Comment:
@bendiknesbo & @hawi74 I think we should refrain from exposing beta functionality in the underlying library. What do you think?

Reply:
I agree, if anything, they should be put under `exp` package, eg. `github.com/coopnorge/go-datadog-lib/v2/exp/metrics`. I don't like mixing stable and unstable APIs in the same package.

Reply:
removing